### PR TITLE
Add in State Slot for Sync Committee Cache

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] In the event of skipped slots, the stored block root does not change. This changes the key of the cache to take account of the slot too. 
- [x] Add a regression test.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/core/helpers/sync_committee.go [hunk 1]
```diff
@@ -186,25 +187,35 @@ func findSubCommitteeIndices(pubKey []byte, pubKeys [][]byte) []types.CommitteeI
 // Retrieve the current sync period boundary root by calculating sync period start epoch
```
Comment (terencechain) [inline beacon-chain/core/helpers/sync_committee.go:187]: Probably want to update the comments here

File: beacon-chain/core/helpers/sync_committee.go [hunk 2]
```diff
@@ -44,7 +45,7 @@ func IsCurrentPeriodSyncCommittee(
 
 		// Fill in the cache on miss.
 		go func() {
-			if err := syncCommitteeCache.UpdatePositionsInCommittee(bytesutil.ToBytes32(root), st); err != nil {
+			if err := syncCommitteeCache.UpdatePositionsInCommittee(root, st); err != nil {
```
Comment (terencechain) [inline beacon-chain/core/helpers/sync_committee.go:48]: Are you sure you don't need to use `combineRootAndSlot` for `UpdatePositionsInCommittee` here?

Comment (terencechain) [reply beacon-chain/core/helpers/sync_committee.go:48]: Ah sorry. The diff didn't show that

File: beacon-chain/core/helpers/sync_committee_test.go [hunk 3]
```diff
@@ -384,3 +386,48 @@ func TestUpdateSyncCommitteeCache_BadRoot(t *testing.T) {
...
+	wantedSlot := params.BeaconConfig().EpochsPerSyncCommitteePeriod.Mul(uint64(params.BeaconConfig().SlotsPerEpoch))
+	assert.NoError(t, state.SetSlot(types.Slot(wantedSlot)))
+	syncCommittee, err = state.CurrentSyncCommittee()
+	assert.NoError(t, err)
+	rand.Shuffle(len(syncCommittee.Pubkeys), func(i, j int) {
+		syncCommittee.Pubkeys[i], syncCommittee.Pubkeys[j] = syncCommittee.Pubkeys[j], syncCommittee.Pubkeys[i]
+	})
+	require.NoError(t, state.SetCurrentSyncCommittee(syncCommittee))
+	// Wait for cache to fill.
+	time.Sleep(100 * time.Millisecond)
```
Comment (prestonvanloon) [inline beacon-chain/core/helpers/sync_committee_test.go:429]: This is concerning. Are caches not atomic?

Comment (prestonvanloon) [inline beacon-chain/core/helpers/sync_committee_test.go:429]:
```suggestion
```

I ran this locally and it passed 100 out of 100 times.

**Review by terencechain — APPROVED** [review]

**Pull request merged**